src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+            assert(last_edge_replacing_input->data.type != SkeletalTrapezoidationEdge::EXTRA_VD);
+            last_edge_replacing_input = insertNode(last_edge_replacing_input, mid, new_node_bead_count);
+            assert(last_edge_replacing_input->data.type != SkeletalTrapezoidationEdge::EXTRA_VD);
+            assert(last_edge_replacing_input->data.isMarked());
+
+
+            debugCheckGraphCompleteness();
+            debugCheckGraphConsistency();
+        }
+    }
+}
+
+SkeletalTrapezoidation::edge_t* SkeletalTrapezoidation::insertNode(edge_t* edge, Point mid, coord_t mide_node_bead_count)
+{
+    edge_t* last_edge_replacing_input = edge;
+
+    graph.nodes.emplace_back(SkeletalTrapezoidationJoint(), mid);
+    node_t* mid_node = &graph.nodes.back();
+
+    edge_t* twin = last_edge_replacing_input->twin;
+    last_edge_replacing_input->twin = nullptr;
+    twin->twin = nullptr;
+    std::pair<SkeletalTrapezoidation::edge_t*, SkeletalTrapezoidation::edge_t*> left_pair
+        = insertRib(*last_edge_replacing_input, mid_node);
+    std::pair<SkeletalTrapezoidation::edge_t*, SkeletalTrapezoidation::edge_t*> right_pair
+        = insertRib(*twin, mid_node);
+    edge_t* first_edge_replacing_input = left_pair.first;
+    last_edge_replacing_input = left_pair.second;
+    edge_t* first_edge_replacing_twin = right_pair.first;
+    edge_t* last_edge_replacing_twin = right_pair.second;
+
+    first_edge_replacing_input->twin = last_edge_replacing_twin;
+    last_edge_replacing_twin->twin = first_edge_replacing_input;
+    last_edge_replacing_input->twin = first_edge_replacing_twin;
+    first_edge_replacing_twin->twin = last_edge_replacing_input;
+
+    mid_node->data.bead_count = mide_node_bead_count;
+
+    return last_edge_replacing_input;
+}
+
+std::pair<SkeletalTrapezoidation::edge_t*, SkeletalTrapezoidation::edge_t*> SkeletalTrapezoidation::insertRib(edge_t& edge, node_t* mid_node)
+{
+    debugCheckGraphConsistency();
+    edge_t* edge_before = edge.prev;
+    edge_t* edge_after = edge.next;
+    node_t* node_before = edge.from;
+    node_t* node_after = edge.to;
+    
+    Point p = mid_node->p;
+
+    std::pair<Point, Point> source_segment = getSource(edge);
+    Point px = LinearAlg2D::getClosestOnLineSegment(p, source_segment.first, source_segment.second);
+    coord_t dist = vSize(p - px);
+    assert(dist > 0);
+    mid_node->data.distance_to_boundary = dist;
+    mid_node->data.transition_ratio = 0; // both transition end should have rest = 0, because at the ends a whole number of beads fits without rest
+
+    graph.nodes.emplace_back(SkeletalTrapezoidationJoint(), px);
+    node_t* source_node = &graph.nodes.back();
+    source_node->data.distance_to_boundary = 0;
+
+    edge_t* first = &edge;
+    graph.edges.emplace_back(SkeletalTrapezoidationEdge());
+    edge_t* second = &graph.edges.back();
+    graph.edges.emplace_back(SkeletalTrapezoidationEdge(SkeletalTrapezoidationEdge::TRANSITION_END));
+    edge_t* outward_edge
```

Comment:
At the time of writing, there isn't really a 'loop below' anymore. Can this be considered done then? (I'm going through the outdated ones too, because many of them are still valid.)

Reply:
Never mind, I just realized that in this and the next few cases, the comment may still be correct, but the code it's referring to may not be the one shown by github.

Reply:
There are a lot of comments that refer to SkeletalTrapezoidation 1629 on Mar 17. I see github doesn't mark them all outdated, but most (all?) of them basically make no sense when you assume they're not.

The code seems to have been placed in Sk..T..Graph.cpp but  that doesn't help much.